dali/operators/imgcodec/image_decoder.h

```diff
@@ -546,9 +545,28 @@ class ImageDecoder : public StatelessOperator<Backend> {
           ParseSample(st->parsed_sample,
                       span<const uint8_t>{static_cast<const uint8_t *>(input_sample.raw_data()),
                                           volume(input_sample.shape())});
-          ROI &roi = rois_[i] = GetRoi(spec_, ws, i, st->parsed_sample.dali_img_info.shape);
-          OutputShape(st->out_shape, st->parsed_sample.dali_img_info, format_,
-                      is_planar_layout, use_orientation_, roi);
+          st->out_shape = st->parsed_sample.dali_img_info.shape;
+          st->out_shape[2] = NumberOfChannels(format_, st->out_shape[2]);
+          if (use_orientation_ &&
+              ((st->parsed_sample.nvimgcodec_img_info.orientation.rotated / 90) & 1)) {
+            std::swap(st->out_shape[0], st->out_shape[1]);
+          }
+          ROI &roi = rois_[i] = GetRoi(spec_, ws, i, st->out_shape);
+          if (roi.use_roi()) {
+            auto roi_sh = roi.shape();
```

Comment:
nitpick/question

Is it possible/handled propely if the roi's end <= begin?

Reply:
it is not supported